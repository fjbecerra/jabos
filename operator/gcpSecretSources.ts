export default function (gcp): any[] {
  return !gcp ? [] : [
    {
      "secret": {
        "name": gcp.secret,
        "items": [
          {
            "key": gcp.serviceAccountKey,
            "path": "gcp_service_account.json"
          }
        ]
      }
    }
  ]
}
