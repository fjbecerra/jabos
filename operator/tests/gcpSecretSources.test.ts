import gcpSecretSources from '../gcpSecretSources';
import { setMock, clearMock } from './settingsMock';

beforeEach(() => {
  setMock();
});

afterEach(() => {
  clearMock();
});

test('gcpSecretSources empty', () => {
  expect(gcpSecretSources(undefined)).toMatchSnapshot();
});

test('gcpSecretSources', () => {
  expect(gcpSecretSources({
    secret: 'secret_value',
    serviceAccountKey: 'serviceAccountKey_value'
  })).toMatchSnapshot();
});
