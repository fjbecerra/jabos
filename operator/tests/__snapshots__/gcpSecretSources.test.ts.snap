// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`gcpSecretSources 1`] = `
Array [
  Object {
    "secret": Object {
      "items": Array [
        Object {
          "key": "serviceAccountKey_value",
          "path": "gcp_service_account.json",
        },
      ],
      "name": "secret_value",
    },
  },
]
`;

exports[`gcpSecretSources empty 1`] = `Array []`;
